src/peergos/shared/storage/RetryStorage.java

```diff
@@ -42,28 +41,30 @@ private int jitter(int minMilliseconds, int rangeMilliseconds) {
     }
 
     private <V> CompletableFuture<V> recurse(int retriesLeft, Supplier<CompletableFuture<V>> f) {
-        Function<Integer, CompletableFuture<V>> compose =
-                i-> {
-                    CompletableFuture<V> res = new CompletableFuture<>();
-                    f.get()
-                            .thenAccept(res::complete)
-                            .exceptionally(e ->  {
-                                if (i == 1) {
-                                    res.completeExceptionally(e);
-                                } else {
-                                    retryAfter(() -> recurse(i - 1, f)
+        CompletableFuture<V> res = new CompletableFuture<>();
```

Comment:
how about moving CF res up a level into runWithRetry and passing it through as a method param?

Reply:
I think it's cleaner to not have references being passed around.

Reply:
ok